### PR TITLE
Fix bugs in IndexedDB wrapper

### PR DESCRIPTION
**Pull request by pwnall**
1) Database.readModifyWrite used an undeclared variable `key`. The intended variable name was `objectKey`.
2) IndexedDB error event handlers inconsistently resolved corresponding promises with errors, instead of rejecting the promises.

### CODE REVIEW

**Comment by pwnall** [comment]
@chasephillips PTAL?

File: third_party/todomvc/vanilla-idb/js/store.js [hunk 1]
```diff
@@ -98,7 +98,7 @@
       return new Promise((resolve, reject) => {
         const transaction = this._idbDatabase.transaction([storeName],
                                                           'readwrite');
-        transaction.onerror = (event) => { resolve(event.target.error); };
+        transaction.onerror = (event) => { reject(event.target.error); };
```
Comment (chasephillips) [inline third_party/todomvc/vanilla-idb/js/store.js]: Thanks for catching and fixing this!

**Review by chasephillips — APPROVED** [review]
LGTM

**Comment by pwnall** [comment]
Thank you for the quick review!

**Pull request merged**